app/models/item.rb

```diff
@@ -118,15 +118,35 @@ def update_stockit_item
     end
   end
 
-  def designate_set_to_stockit_order(order_id)
+  def update_designation(params)
     inventory_packages.set_items.each do |package|
-      package.designate_to_stockit_order(order_id)
+      orders_package_with_package_id = OrdersPackage.find_packages(package.order_id, package.id)
+      orders_package_with_params_id = OrdersPackage.find_packages(params[:order_id], package.id)
+      OrdersPackage.update_designation(orders_package_with_package_id, params[:order_id])
+      OrdersPackage.delete_unwanted_cancelled_packages(orders_package_with_params_id, params[:order_id])
```

Comment:
@Darshan-Chauhan there are a lot of class methods here... I wonder if they would be better off as instance methods on `OrdersPackage`